js/xt.js

```diff
@@ -0,0 +1,1072 @@
+        //         "multiplierUp": "0.005",
+        //         "maxOpenOrders": 200,
+        //         "maxEntrusts": 120,
+        //         "makerFee": "0.00015",
+        //         "takerFee": "0.0007",
+        //         "liquidationFee": "0.015",
+        //         "marketTakeBound": "0.005",
+        //         "depthPrecisionMerge": 5,
+        //         "labels": [
+        //         "HOT"
+        //         ],
+        //         "onboardDate": 1654857001000,
+        //         "enName": "BTCUSDT Perpetual",
+        //         "cnName": "BTCUSDT 永续",
+        //         "minStepPrice": "0.5"
+        //       },
+        //    ]
+        //
+        // COIN
+        //
+        //  [
+        //    {
+        //      "symbol": "btc_usd",
+        //      "contractType": "PERPETUAL",
+        //      "underlyingType": "COIN_BASED",
+        //      "contractSize": "100",
+        //      "tradeSwitch": true,
+        //      "state": 0,
+        //      "initLeverage": 10,
+        //      "initPositionType": "CROSSED",
+        //      "baseCoin": "btc",
+        //      "quoteCoin": "usd",
+        //      "baseCoinPrecision": 8,
+        //      "baseCoinDisplayPrecision": 8,
+        //      "quoteCoinPrecision": 8,
+        //      "quoteCoinDisplayPrecision": 8,
+        //      "quantityPrecision": 0,response
+        //      "maxEntrusts": 120,
+        //      "makerFee": "0.00015",
+        //      "takerFee": "0.0007",
+        //      "liquidationFee": "0.015",
+        //      "marketTakeBound": "0.005",
+        //      "depthPrecisionMerge": 5,
+        //      "labels": [
+        //      "HOT"
+        //      ],
+        //      "onboardDate": 1654859400000,
+        //      "enName": "BTCUSD Perpetual",
+        //      "cnName": "BTCUSD 永续",
+        //      "minStepPrice": "0.5"
+        //    },
+        //  ]
+        //
+        const marketsUSDT = await this.parseSwapAndFutureMarkets (responseUSDT);
+        const marketsCOIN = await this.parseSwapAndFutureMarkets (responseCOIN);
+        return this.arrayConcat (marketsUSDT, marketsCOIN);
+    }
+
+    async parseSwapAndFutureMarkets (response) {
+        const result = [];
+        const data = this.safeValue (response, 'result', []);
+        for (let i = 0; i < data.length; i++) {
+            const market = data[i];
+            const id = this.safeString (market, 'symbol');
+            const baseId = this.safeString (market, 'baseCoin');
+            const quoteId = this.safeString (market, 'quoteCoin');
+            const base = this.safeCurrencyCode (baseId);
+            const quote = this.safeCurrencyCode (quoteId);
+            const underlyingType = this.safeString (market, 'underlyingType', '');
+            let symbol = base + '/' + quote;
+            let linear = undefined;
+            let inverse = undefined;
+            if (underlyingType === 'U_BASED') {
+                symbol += ':USDT';
```

Comment:
We should avoid hardcoded values at all costs, what if the exchange in the future introduces a linear USDC linear swap? This code wouldn't work, so I think we should be using here the `quoteCoin` as the settlement